### ISSUE TITLE
Query with sub-query doesn't use the index on the outer query.

### ISSUE THREAD

**Issue by dalegarv** [orientechnologies/orientdb#7329]
## Expected behavior  
Query with sub-query should use the index on the outer query.

## Actual behavior  
The outer query is not using the index. 

## Steps to reproduce  
select from Grant where domainId = 'HRVANTAGE' and actor IN (SELECT securityGroup FROM MemberOf WHERE member = #89:286)

Explain shows that the MemberOf.member index is being used on the sub-query, but neither the Grant.domainId_actor nor the Grant.actor indexes on the outer query are being used, so the query is slow and gets the "fetched more than 50000 records" warning.

I've tried assigning the results of the sub-query to a variable using LET, but got the same behavior.  The only solution I could find was to run the sub-query first, and then run the outer query separately passing the results from the first query.

**Comment by luigidellaquila** [comment]
Hi @dalegarv 

This is a known limitation in v 2.2.x, in v 3.0 we will have a much smarter execution planner.
What you can do now is use a batch script instead of a single query, eg.

```
LET $a = SELECT securityGroup FROM MemberOf WHERE member = #89:286;
select from Grant where domainId = 'HRVANTAGE' and actor IN $a;
```

In this case it will use the index

Thanks

Luigi

**Comment by lvca** [comment]
Implemented in 3.0
